spec/iot-js-spec.md

```diff
@@ -123,31 +118,29 @@ dictionary OicDeviceInfo {
 interface OicClient: EventTarget {
   // client API: discovery; Promise succeeds when request is successfully sent
   Promise<void> findResources(OicDiscoveryOptions options);  // GET oic/res
-  Promise<void> findDevices(OicDiscoveryOptions options);  // GET /oic/d
+  Promise<void> findDevices();  // GET /oic/d
 
   // client API: CRUDN
-  Promise<OicResource> createResource(OicResourceInit resource);  // create remote res
-  Promise<OicResource> retrieveResource(USVString resourceId);  // get remote res
-  Promise<void> updateResource(USVString resourceId, OicResourceInit resource);  // post or put remote res
-  Promise<void> deleteResource(USVString resourceId);  // delete remote res
-  Promise<OicResource> startObserving(USVString resourceId);  // retrieve + observe flag
-  Promise<void> cancelObserving(USVString id);
-
-  attribute EventHandler<OicResourceChangeEvent> onresourcechange;  // observe
```

Comment:
OK, if all our target platforms support custom actions during event handler attachment/detachment, then I'm OK with an event-only approach. Nevertheless, now that I think about it, I believe that if we do onresourceupdate, it's only possible in node.js if we add the event to OicResource. The reason is that yes, I can do something special when the first event handler is attached, and yes, I can do something special when the last event handler is detached, but I can only do something special concerning the object to which it is attached/detached. If that object is device.client, I won't know which resource to start observing or which resource to stop observing. So, the event would have to be attached to an OicResource for this to work.